### PR TITLE
Draw box

### PR DESCRIPTION
**Pull request by Unnunu**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.28%) on `us`.

⚠️ This PR introduces a warning:
- src/msg.c:2265: warning: passing arg 15 of `draw_box' from incompatible pointer type

File: src/draw_box.c [hunk 1]
```diff
@@ -0,0 +1,830 @@
...
+|-6.0  0.0  0.0  0.0|
+*/
+Mtx gBoxMatrix = {
+    .m = {
+        // integer portion
+        { 0x00000000, 0x00000000,
+          0x00000000, 0x00000000 },
+        { 0x00000000, 0x00000000,
+          0xFFFA0000, 0x00000000 },
+        //fractional portion
```
Comment (nanaian) [inline src/draw_box.c:333]:
```suggestion
        // fractional portion
```

File: src/draw_box.c [hunk 2]
```diff
@@ -0,0 +1,830 @@
...
+        { 0x00000000, 0x00000000,
+          0x00000000, 0x00000000 },
+        { 0x00000000, 0x00000000,
+          0x00000000, 0x00000000 }
+    }
+};
+
+#define _INTEGER_LOG2(x) ((x) <= 1 ? 0 : (x) <= 2 ? 1 : (x) <= 4 ? 2 : (x) <= 8 ? 3 : (x) <= 16 ? 4 : (x) <= 32 ? 5 : (x) <= 64 ? 6 : (x) <= 128 ? 7 : (x) <= 256 ? 8 : (x) <= 512 ? 9 : 10)
```
Comment (nanaian) [inline src/draw_box.c:341]: Can you move this to the top of the file & remove the leading `_`

File: src/draw_box.c [hunk 3]
```diff
@@ -0,0 +1,830 @@
...
+                                        G_TX_CLAMP, G_TX_CLAMP, 9, 3, G_TX_NOLOD, G_TX_NOLOD);
+                        gDPSetTileSize(gMasterGfxPos++, G_TX_RENDERTILE, (160 - width / 2) * 4, 0, ((160 - width / 2) + width - 1) * 4, lineHeight * 4);
+                        gSPVertex(gMasterGfxPos++, &vtx_drawbox1, 4, 0);
+                        gSP2Triangles(gMasterGfxPos++, 0, 3, 1, 0, 0, 2, 3, 0);
+                        gDPPipeSync(gMasterGfxPos++);
+                        gSPMatrix(gMasterGfxPos++, &gBoxMatrix, G_MTX_NOPUSH | G_MTX_MUL | G_MTX_MODELVIEW);
+                    }
+                }
+            }
+            else {
```
Comment (nanaian) [inline src/draw_box.c:818]:
```suggestion
            } else {
```

**Review by nanaian — CHANGES_REQUESTED** [review]

**Pull request merged**